uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/BAO/LeaveRequestTest.php

```diff
@@ -129,6 +129,57 @@ public function testUpdatingALeaveRequestShouldNotDuplicateTheLeaveRequestDates(
     $this->assertEquals($toDate->format('Y-m-d'), $dates[1]->date);
   }
 
+  public function testUpdatingALeaveRequestShouldNotThrowOverLappingLeaveRequestExceptionWhenTheDatesOverlap()
+  {
+
+    $period = AbsencePeriodFabricator::fabricate([
+      'start_date' => CRM_Utils_Date::processDate('2016-01-01'),
+      'end_date'   => CRM_Utils_Date::processDate('2016-12-31'),
+    ]);
+
+    $periodEntitlement = LeavePeriodEntitlementFabricator::fabricate([
+      'type_id' => $this->absenceType->id,
+      'contact_id' => 1,
+      'period_id' => $period->id
+    ]);
+
+    $this->createLeaveBalanceChange($periodEntitlement->id, 20);
+
+    $workPattern = WorkPatternFabricator::fabricateWithA40HourWorkWeek();
+    ContactWorkPatternFabricator::fabricate([
+      'contact_id' => 1,
+      'pattern_id' => $workPattern->id
+    ]);
+    $leaveRequestStatuses = array_flip(LeaveRequest::buildOptions('status_id'));
+
+    $fromDate1 = new DateTime('2016-11-02');
+    $toDate1 = new DateTime('2016-11-04');
+    $leaveRequest1 = LeaveRequestFabricator::fabricateWithoutValidation([
+      'type_id' => $this->absenceType->id,
+      'contact_id' => 1,
+      'status_id' => $leaveRequestStatuses['Waiting Approval'],
+      'from_date' => $fromDate1->format('YmdHis'),
+      'from_date_type' => 1,
+      'to_date' => $toDate1->format('YmdHis'),
+      'to_date_type' => 1
+    ], true);
+
+    //updating leave request
+    $fromDate = new DateTime('2016-11-03');
+    $toDate = new DateTime('2016-11-05');
+    $leaveRequest2 = LeaveRequest::create([
+      'id' => $leaveRequest1->id,
+      'type_id' => $this->absenceType->id,
+      'contact_id' => 1,
+      'status_id' => 1,
+      'from_date' => $fromDate->format('YmdHis'),
```

Comment:
you're not using fromDate and toDate anywhere else, so why not do this instead?

```php
 'from_date' => CRM_Utils_Date::processDate('2016-11-03'),
 'to_date' => CRM_Utils_Date::processDate('2016-11-05'),
```

You could do the same to fromDate1 and toDate1